core/interfaces/rebase.py

```diff
@@ -284,6 +296,95 @@ def base_ref(self):
 
         return base_ref
 
+    def nearest_branch(self, branch, default="master"):
+        """
+        Find the nearest commit in current branch history that exists
+        on a different branch and return that branch name.
+
+        We filter these branches through a list of known ancestors which have
+        an initial branch point with current branch, and pick the first one
+        that matches both.
+
+        If no such branch is found, returns the given default ("master" if not
+        specified).
+
+        Solution snagged from:
+        http://stackoverflow.com/a/17843908/484127
+        http://stackoverflow.com/questions/1527234
+        """
+        try:
+            branch_tree = self.git("show-branch", "--no-color").splitlines()
+        except GitSavvyError:
+            return default
+
+        util.debug.add_to_log('nearest_branch: found %s show-branch results' %
+                              len(branch_tree))
+
+        relatives = []
+        for rel in branch_tree:
+            match = re.search(self.nearest_node_pattern, rel)
+            # print(match.groups(), branch, default)
+            if not match:
+                # util.debug.add_to_log('Could not parse relative string %s' % rel)
+                continue
+            branch_name = match.groups()[0]
+            if branch_name != branch and branch_name not in relatives:
+                relatives.append(branch_name)
+
+        util.debug.add_to_log('nearest_branch: found %s relatives: %s' %
+                              (len(relatives), relatives))
+
+        if not relatives:
+            util.debug.add_to_log('No relatives found. Possibly on a root branch!')
+            return default
+
+        util.debug.add_to_log('nearest_branch: filter out branches that do not'
+                              ' share branch-out nodes')
+        diff = difflib.Differ()
+        branch_commits = self.git("rev-list", "--first-parent", branch).splitlines()
+        nearest = None
+        max_iterations = 100
+        for relative in relatives:
+            util.debug.add_to_log('nearest_branch: Getting common commits with %s' % relative)
+            relative_commits = self.git("rev-list", "--first-parent", relative).splitlines()
```

Comment:
You may set the max iterations by `self.git("rev-list", "-{}".format(max_iterations), "--first-parent", relative)`